src/core/transport/chttp2_transport.c

```diff
@@ -1513,7 +1552,29 @@ void grpc_chttp2_incoming_byte_stream_push(grpc_exec_ctx *exec_ctx,
 }
 
 void grpc_chttp2_incoming_byte_stream_finished(
-    grpc_exec_ctx *exec_ctx, grpc_chttp2_incoming_byte_stream *bs) {
+    grpc_exec_ctx *exec_ctx, grpc_chttp2_incoming_byte_stream *bs, int success,
+    int from_parsing_thread) {
+  if (!success) {
+    if (from_parsing_thread) {
+      gpr_mu_lock(&bs->transport->mu);
+    }
+    grpc_exec_ctx_enqueue(exec_ctx, bs->on_next, 0);
```

Comment:
grpc_exec_ctx_enqueue accepts a null closure (and does nothing) -- since
it's awfully convenient.

On Mon, Dec 14, 2015 at 3:36 PM Sree Kuchibhotla <notifications@github.com>
wrote:

> In src/core/transport/chttp2_transport.c
> <https://github.com/grpc/grpc/pull/4453#discussion_r47576693>:
>
> > @@ -1513,7 +1552,29 @@ void grpc_chttp2_incoming_byte_stream_push(grpc_exec_ctx *exec_ctx,
> >  }
> >
> >  void grpc_chttp2_incoming_byte_stream_finished(
> > -    grpc_exec_ctx *exec_ctx, grpc_chttp2_incoming_byte_stream *bs) {
> > +    grpc_exec_ctx *exec_ctx, grpc_chttp2_incoming_byte_stream *bs, int success,
> > +    int from_parsing_thread) {
> > +  if (!success) {
> > +    if (from_parsing_thread) {
> > +      gpr_mu_lock(&bs->transport->mu);
> > +    }
> > +    grpc_exec_ctx_enqueue(exec_ctx, bs->on_next, 0);
>
> Is it guaranteed that bs->on_next would be non-null here ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/4453/files#r47576693>.
>